File: pratica2/localizar_cursor.m
%{

Program: [localizar_cursor]

Author: [Matheus Santos Silva Pereira]

Date: [13.03.2024]

Description: [Localizar o cursor, calcular area do circulo e retangulo]

License: [GNU General Public License]

%}

% Definição da função para calcular os valores do círculo
function [comprimento, area] = Circulo(raio)
    % Cálculo do comprimento da circunferência e área do círculo
    comprimento = 2 * pi * raio;
    area = pi * raio * raio;
end

% Definição da função para calcular os valores do retângulo
function [area, perimetro] = Retangulo(comprimento, largura)
    % Cálculo da área e do perímetro do retângulo
    area = comprimento * largura;
    perimetro = 2 * (comprimento + largura);
end

% Função para localizar o cursor
function quadrant = localizarCursor(Width, Height)
    cursorPos = get(0, 'PointerLocation');
    x = cursorPos(1);
    y = cursorPos(2);
    
    if x <= Width && y <= Height
        quadrant = 'A';
    elseif x >= Width && y <= Height
        quadrant = 'B';
    elseif x <= Width && y >= Height
        quadrant = 'C';
    elseif x >= Width && y >= Height
        quadrant = 'D';
    else
        quadrant = '';
    end
end

% Código principal
clc; clear; close all;

continua = 's'; % Variável para continuar ou não o programa
Width = 960; Height = 540;

while continua == 's'
    fprintf('\n');
    fprintf('SELECIONE UMA OPCAO:\n');
    fprintf('1 - Localizar cursor\n');
    fprintf('2 - Calcular circulo\n');
    fprintf('3 - Calcular retangulo\n');
    fprintf('Pressione ESC para sair\n');
    
    if kbhit()
        tecla = getkey();
        if tecla == 27
            fprintf('Programa finalizado.\n');
            break;
        end
    end
    
    escolha = input('');
    
    switch escolha
        case 1
            quadrant = localizarCursor(Width, Height);
            switch quadrant
                case 'A'
                    fprintf('O ponteiro do mouse esta localizado no quadrante A\n');
                case 'B'
                    fprintf('O ponteiro do mouse esta localizado no quadrante B\n');
                case 'C'
                    fprintf('O ponteiro do mouse esta localizado no quadrante C\n');
                case 'D'
                    fprintf('O ponteiro do mouse esta localizado no quadrante D\n');
                otherwise
                    fprintf('O ponteiro do mouse esta localizado fora dos quadrantes\n');
            end
        case 2
            raio = input('Digite o raio do circulo: ');
            [comprimento, area] = Circulo(raio);
            fprintf('Comprimento da circunferencia: %.2f\n', comprimento);
            fprintf('Area do circulo: %.2f\n', area);
        case 3
            comprimento = input('Digite o comprimento do retangulo: ');
            largura = input('Digite a largura do retangulo: ');
            [area, perimetro] = Retangulo(comprimento, largura);
            fprintf('Perimetro do retangulo: %.2f\n', perimetro);
            fprintf('Area do retangulo: %.2f\n', area);
        otherwise
            fprintf('Opcao invalida\n');
    end
end

File: pratica2/localizar_cursor.cpp
/*

* Program: [localizar_cursor]

* Author: [Matheus Santos Silva Pereira]

* Date: [13.03.2024]

* Description: [Localizar o cursor, calcular area do circulo e retangulo]

* License: [GNU General Public License]

*/

#include <iostream>
#include <windows.h>
#include <conio.h>

using namespace std;

typedef struct {
    float raio;
    float comprimento;
    float area;
} Circle;

typedef struct {
    float comprimento;
    float largura;
    float area;
    float perimetro;
} Rectangle;

#define PI 3.14

void Circulo(Circle *c) {
    cout << "Digite o raio do circulo: ";
    cin >> c->raio;
    
    c->comprimento = 2 * PI * c->raio;
    c->area = PI * c->raio * c->raio;
}

void Retangulo(Rectangle *r) {
    cout << "Digite o comprimento do retangulo: ";
    cin >> r->comprimento;
    
    cout << "Digite a largura do retangulo: ";
    cin >> r->largura;
    
    r->area = r->comprimento * r->largura;
    r->perimetro = 2 * (r->comprimento + r->largura);
}

char localizarCursor(int Width, int Height) {
    POINT cursorPos;
    GetCursorPos(&cursorPos);
    int x = cursorPos.x; 
    int y = cursorPos.y;
    
    if (x <= Width && y <= Height){
        return 'A'; 
    } else if (x >= Width && y <= Height){
        return 'B';
    } else if (x <= Width && y >= Height){
        return 'C';
    } else if (x >= Width && y >= Height){
        return 'D';
    } else {
        return '\0';
    }
}

int main() {
    int escolha;
    char continua = 's';
    int Width = 960, Height = 540;
    
    while (continua == 's') {
        cout << "\n";
        cout << "SELECIONE UMA OPCAO:\n";
        cout << "1 - Localizar cursor\n";
        cout << "2 - Calcular circulo\n";
        cout << "3 - Calcular retangulo\n";
        cout << "Pressione ESC para sair\n";
        
        if (_kbhit()) {
            char tecla = _getch();
            if (tecla == 27) {
                cout << "Programa finalizado.\n";
                break;
            }
        }
        
        cin >> escolha;
        
        switch(escolha) {
            case 1: {
                char quadrant = localizarCursor(Width, Height);
                if(quadrant == 'A') {
                    cout << "O ponteiro do mouse esta localizado no quadrante A\n";
                } else if(quadrant == 'B') {
                    cout << "O ponteiro do mouse esta localizado no quadrante B\n";
                } else if(quadrant == 'C') {
                    cout << "O ponteiro do mouse esta localizado no quadrante C\n";
                } else if(quadrant == 'D') {
                    cout << "O ponteiro do mouse esta localizado no quadrante D\n";
                } else {
                    cout << "O ponteiro do mouse esta localizado fora dos quadrantes\n";
                }
                break;
            }
            case 2: {
                Circle c;
                for (int i = 0; i < 3; ++i) {
                    Circulo(&c);
                    cout << "Comprimento da circunferencia: " << c.comprimento << endl;
                    cout << "Area do circulo: " << c.area << endl;
                }
                break;
            }
            case 3: {
                Rectangle r;
                for (int i = 0; i < 3; ++i) {
                    Retangulo(&r);
                    cout << "Perimetro do retangulo: " << r.perimetro << endl;
                    cout << "Area do retangulo: " << r.area << endl;
                }
                break;
            }
            default:
                cout << "Opcao invalida\n";
                break;
        }
    }
    
    return 0;
}
